### PR TITLE
Additional gsuite rules

### PR DESCRIPTION
**Pull request by nhakmiller**
### Background

Adds fifteen new gsuite rules.

### Changes

* Adds fifteen new gsuite rules

### Testing

* Unit testing

### CODE REVIEW

File: gsuite_reports_rules/gsuite_drive_overly_visible.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        if (details.get('type') == 'access' and
+                details.get('name') in RESOURCE_CHANGE_EVENTS and
+                details.get('visibility') in PERMISSIVE_VISIBILITY):
+            return True
+
+    return False
+
+
+def dedup(event):
+    return event['p_row_id']
```
Comment (nhakmiller) [inline gsuite_reports_rules/gsuite_drive_overly_visible.py:28]: This is simpler than extracting the document ID from the list of events.

File: gsuite_reports_rules/gsuite_google_access.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+def rule(event):
```
Comment (nhakmiller) [inline gsuite_reports_rules/gsuite_google_access.py:1]: Couldn't think of a good dedup/title function here, probably doesn't need one. The event is indicating that a google employee is looking at your documents for some reason. There are more details in the gsuite super admin console.

File: global_helpers/panther_base_helpers.py [hunk 3]
```diff
@@ -58,3 +58,29 @@ def is_dmz_tags(resource):
...
+    'messageValue',
+    'multiMessageValue',
+]
+
+
+# GSuite parameters are formatted as a list of dictionaries, where each dictionary has a 'name' key
+# that maps to the name of the parameter, and one key from GSUITE_PARAMETER_VALUES that maps to the
+# value of the parameter. This means to lookup the value of a particular parameter, you must
+# traverse the entire list of parameters to find it and then know (or guess) what type of value it
+# contains. This helper function handles that for us.
```
Comment (austinbyers) [inline global_helpers/panther_base_helpers.py:77]: Can you maybe add an example of what the input/output should be? It's hard to picture

File: gsuite_reports_rules/gsuite_doc_ownership_transfer.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+def rule(event):
+    if event['id'].get('applicationName') != 'admin':
+        return False
+
+    for details in event.get('events', [{}]):
+        if (details.get('type') == 'DOCS_SETTINGS' and
+                details.get('name') == 'TRANSFER_DOCUMENT_OWNERSHIP'):
+            new_owner = param_lookup(details.get('parameters', {}), 'NEW_VALUE')
+            return bool(new_owner) and not any(
+                [new_owner.endswith(x) for x in ORG_DOMAINS])
```
Comment (austinbyers) [inline gsuite_reports_rules/gsuite_doc_ownership_transfer.py:17]: Don't need the list construction:

```suggestion
            return bool(new_owner) and not any(
                new_owner.endswith(x) for x in ORG_DOMAINS)
```

File: global_helpers/panther_base_helpers.py (hunk 3 above)
Comment (nhakmiller) [reply global_helpers/panther_base_helpers.py:77]: Sure, I added an example. Let me know if that clears things up a bit.

Comment (austinbyers) [reply global_helpers/panther_base_helpers.py:77]: Definitely

**Pull request merged**